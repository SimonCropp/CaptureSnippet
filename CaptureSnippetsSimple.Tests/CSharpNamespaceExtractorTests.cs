using System.Collections.Generic;
using System.IO;
using System.Linq;
using CaptureSnippets;
using ObjectApproval;
using Xunit;

public class CSharpNamespaceExtractorTests : TestBase
{
    [Fact]
    public void CanExtractUsingsFromCSharpLanguage()
    {
        var input = @"
  #region UsingsSection
  using System;
  #endregion";
        var snippets = FromText(input);
        ObjectApprover.VerifyWithJson(snippets);
    }

    [Fact]
    public void WithNameSpaceThatHasKeywordInIt()
    {
        var input = @"
  #region UsingsSection
  using LibraryUsingSystem;
  #endregion";
        var snippets = FromText(input);
        ObjectApprover.VerifyWithJson(snippets);
    }

    [Fact]
    public void WithSnippetIncludingKeywordInComments()
    {
        var input = @"
  #region UsingsSection
//using of System;
using System;
  #endregion";
        var snippets = FromText(input);
        ObjectApprover.VerifyWithJson(snippets);
    }

    [Fact]
    public void WithNamespaceThatIncludesAlias()
    {
        var input = @"
  #region UsingsSection
using Math = System.Math;
  #endregion";
        var snippets = FromText(input);
        ObjectApprover.VerifyWithJson(snippets);
    }

    [Fact]
    public void WithDuplicatedUsings()
    {
        var input = @"
  #region UsingsSection
using System;
using System.Core;
using System;
  #endregion";
        var snippets = FromText(input);
        ObjectApprover.VerifyWithJson(snippets);
    }

    [Fact]
    public void WithIncludesOutOfSnippetRegion()
    {
        var input = @"
using System;
using System.Core;

  #region TestClass
  public class Test { }
  #endregion";
        var snippets = FromText(input);
        ObjectApprover.VerifyWithJson(snippets);
    }

    [Fact]
    public void WithOutOfOrderIncludes()
    {
        var input = @"
using System.Core;
using System;
using System.Xml.Linq;

  #region TestClass
  public class Test { }
  #endregion";
        var snippets = FromText(input);
        ObjectApprover.VerifyWithJson(snippets);
    }

    List<Snippet> FromText(string contents)
    {
        var extractor = FileSnippetExtractor.Build("package");
        using (var stringReader = new StringReader(contents))
        {
            return extractor.AppendFromReader(stringReader, "path.cs").ToList();
        }
    }
}