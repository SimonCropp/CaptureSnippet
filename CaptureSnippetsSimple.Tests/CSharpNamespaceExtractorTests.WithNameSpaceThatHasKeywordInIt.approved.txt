[
  {
    "Key": "usingssection",
    "Language": "cs",
    "Value": "using LibraryUsingSystem;",
    "Includes": [
      "LibraryUsingSystem"
    ],
    "Error": null,
    "FileLocation": "path.cs(3-4)",
    "IsInError": false
  }
]